Mage/src/main/java/mage/abilities/common/EntersBattlefieldOneOrMoreTriggeredAbility.java

```diff
@@ -40,37 +40,32 @@ private EntersBattlefieldOneOrMoreTriggeredAbility(final EntersBattlefieldOneOrM
 
     @Override
     public boolean checkEventType(GameEvent event, Game game) {
-        return event.getType() == GameEvent.EventType.ZONE_CHANGE_GROUP;
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE_BATCH;
     }
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        ZoneChangeGroupEvent zEvent = (ZoneChangeGroupEvent) event;
+
         Player controller = game.getPlayer(this.controllerId);
-        if (zEvent.getToZone() != Zone.BATTLEFIELD || controller == null) {
+        if (controller == null) {
             return false;
         }
 
+        ZoneChangeBatchEvent zEvent = (ZoneChangeBatchEvent) event;
+        Stream<Permanent> enteringPermanents = zEvent.getEvents().stream()
+                .filter(z -> z.getToZone() == Zone.BATTLEFIELD)
+                .map(ZoneChangeEvent::getTarget)
+                .filter(Objects::nonNull)
+                .filter(permanent -> filterPermanent.match(permanent, this.controllerId, this, game));
+
         switch (this.targetController) {
             case YOU:
-                if (!controller.getId().equals(zEvent.getPlayerId())) {
-                    return false;
-                }
-                break;
+                return enteringPermanents.anyMatch(permanent -> permanent.getControllerId().equals(this.controllerId));
```

Comment:
Use getControllerId() and getTargetController() instead direct access due good code style.

Reply:
`targetController` is fine to access directly, it's internal to this class. Still need to update to `getControllerId()` though